Add preview for mail attachment

GretaD opened the pull request:
fixes #5701 

- [x] fit the full view on the screen, no scrolling
- [x] fit the mobile view on the screen, no scrolling

ChristophWurst commented:
* [ ] BUG the modal is shown as soon as just one of the attachments is a picture. It should only be available to an attachment that is an image. Nothing else.

![Bildschirmfoto von 2021-12-27 11-23-24](https://user-images.githubusercontent.com/1374172/147462591-a4058840-753a-4919-9573-a56b01aba7e6.png)

^ tested with pic & md file

ChristophWurst commented:
* [ ] BUG images are cropped. Try sending a portrait mode image. It will show landscape with the bottom cropped off

GretaD commented:
> * [ ]  BUG images are cropped. Try sending a portrait mode image. It will show landscape with the bottom cropped off

i cannot reproduce this one, but im assuming its because the size used to be isMobile: full ? large. I changed it into large only and added overflow scroll. Can you please test, or send me that image you could reproduce

GretaD commented:
> > * [ ]   BUG images are cropped. Try sending a portrait mode image. It will show landscape with the bottom cropped off
> 
> i cannot reproduce this one, but im assuming its because the size used to be isMobile: full ? large. I changed it into large only and added overflow scroll. Can you please test, or send me that image you could reproduce

i can reproduce and the scrolling fixed it

GretaD commented on src/components/AttachmentImageViewer.vue:
```diff
@@ -0,0 +1,33 @@
+		Modal,
+	},
+
+	mixins: [isMobile],
```
leftover

GretaD commented:
![desktopatt](https://user-images.githubusercontent.com/12728974/147667929-df28ada0-9f05-40e7-b04a-94e2ed06564c.png)

mobile
![mobileatt](https://user-images.githubusercontent.com/12728974/147667938-7b435c86-c89a-4d0e-81e8-536208b70016.png)

ChristophWurst commented:
> ![desktopatt](https://user-images.githubusercontent.com/12728974/147667929-df28ada0-9f05-40e7-b04a-94e2ed06564c.png)
> 
> mobile ![mobileatt](https://user-images.githubusercontent.com/12728974/147667938-7b435c86-c89a-4d0e-81e8-536208b70016.png)

look at the pictures. they are still cropped.

st3iny reviewed (commented):
The overflow of tall pictures is still not fixed. I manually added `overflow: auto` to `.modal-container` and it worked fine (showed a vertical scroll bar).

st3iny commented on src/components/AttachmentImageViewer.vue:
```diff
@@ -0,0 +1,50 @@
+		box-shadow: none !important;
+		display: flex !important;
+		width: 90vw;
+	}
```
```suggestion
	}
	.modal-container {
		overflow: auto !important;
	}
```

GretaD commented:
> The overflow of tall pictures is still not fixed. I manually added `overflow: auto` to `.modal-container` and it worked fine (showed a vertical scroll bar).

the thing is that it should fit the screen and not scroll. I fixed it with scrolling but than i checked how viewer does and it doesnt scroll.

GretaD commented:
please give it a final test. the mobile version of preview its a bit small but readable. I cannot find a solution to fit the screen for portrait and not brake images that are horizontally long. E.g
![testhorixontalll](https://user-images.githubusercontent.com/12728974/148059400-616d9bc1-79ed-4471-a23c-f2d51eb202de.png)

st3iny reviewed (approved):
Images inside modals were still cropped but I deployed a fix.
